src/engraving/dom/guitarbend.cpp

```diff
@@ -417,10 +576,97 @@ GuitarBend* GuitarBend::findPrecedingBend() const
     }
 
     GuitarBend* precedingBend = startN->bendBack();
-    if (precedingBend && precedingBend->type() != GuitarBendType::SLIGHT_BEND) {
+    if (precedingBend && precedingBend->isDive() == isDive()
+        && precedingBend->bendType() != GuitarBendType::SLIGHT_BEND
+        && precedingBend->bendType() != GuitarBendType::DIP
+        && precedingBend->bendType() != GuitarBendType::SCOOP) {
         return precedingBend;
     }
 
+    if (bendType() == GuitarBendType::PRE_DIVE) {
+        ChordRest* prevCR = prevChordRest(startN->chord());
+        if (prevCR && prevCR->isRest()) {
+            WhammyBar* whammyBar = findOverlappingWhammyBar(prevCR->tick(), tick2());
+            if (whammyBar) {
+                while (prevCR && prevCR->isRest() && prevCR->tick() > whammyBar->tick()) {
+                    prevCR = prevChordRest(prevCR);
+                }
+            }
+        }
+        if (!prevCR || !prevCR->isChord()) {
+            return nullptr;
+        }
+        Chord* prevChord = toChord(prevCR);
+        for (Note* note : prevChord->notes()) {
+            while (note->tieBack() && note->tieBack()->startNote()) {
+                note = note->tieBack()->startNote();
+            }
+            GuitarBend* precedingBend = note->bendBack();
```

Comment:
see MSVC warning: `declaration of 'precedingBend' hides previous local declaration`